# Activity 3 - K-means clustering with Pew Data

#### Reading data from CSV

In [1]:
pew <- read.csv("June 10-July 12, 2015 - Gaming, Jobs and Broadband - CSV.csv")

#### Extracting required data and cleaning it

In [2]:
data <- data.frame(Age = pew$age, Date=pew$date1a, Game = pew$game1, Empl = pew$emplnw, Education = pew$educ2)
clean_data <- na.omit(data)
clean_data <- clean_data[!clean_data$Date %in% c(8,9, NA),]
clean_data <- clean_data[!clean_data$Game %in% c(8,9, NA),]
clean_data <- clean_data[!clean_data$Empl %in% c(98,99, NA),]
clean_data <- clean_data[!clean_data$Age %in% c(98,99, NA),]
clean_data <- clean_data[!clean_data$Education %in% c(98,99, NA),]
head(clean_data)

Age Date Game Empl Education
1 47  2    2    4    6        
2 63  2    1    3    4        
4 40  2    2    1    5        
5 65  2    1    3    4        
6 69  2    2    2    6        
9 77  2    2    3    6

#### Scaling the cleaned data

In [3]:
scaled_data<- data.frame(scale(clean_data))
scaled_data

Age         Date       Game       Empl       Education  
1    -0.09555574  0.3853757  1.0267677  1.3367497  0.45235747
2     0.81636738  0.3853757 -0.9733521  0.6052060 -0.55988999
4    -0.49452211  0.3853757  1.0267677 -0.8578816 -0.05376626
5     0.93035777  0.3853757 -0.9733521  0.6052060 -0.55988999
6     1.15833855  0.3853757  1.0267677 -0.1263378  0.45235747
9     1.61430011  0.3853757  1.0267677  0.6052060  0.45235747
11    1.84228089  0.3853757  1.0267677  0.6052060  0.45235747
12    1.61430011  0.3853757 -0.9733521  0.6052060 -1.06601373
13    0.18942023  0.3853757 -0.9733521 -0.8578816 -0.55988999
14    1.44331452  0.3853757 -0.9733521  0.6052060 -0.05376626
15   -0.32353652  0.3853757 -0.9733521 -0.8578816  0.45235747
16   -0.26654133  0.3853757  1.0267677 -0.8578816 -0.55988999
18    2.18425206  0.3853757  1.0267677  0.6052060 -0.05376626
19    1.10134335  0.3853757  1.0267677  0.6052060  1.46460494
20    0.53139140  0.3853757  1.0267677 -0.8578816  0.45235747
21   -1.00747886  0.3853757 -0.9733521 -0.8578816 -1.06601373
25    1.21533374  0.3853757  1.0267677  0.6052060  1.46460494
26   -0.72250289  0.3853757  1.0267677  1.3367497  1.46460494
27   -0.55151730  0.3853757 -0.9733521  1.3367497  0.45235747
28    2.07026167  0.3853757  1.0267677  0.6052060 -1.06601373
29    1.55730491  0.3853757  1.0267677  0.6052060  0.95848121
30    0.81636738  0.3853757  1.0267677 -0.8578816  1.46460494
31    0.98735296  0.3853757 -0.9733521  0.6052060  1.46460494
32    0.47439621  0.3853757  1.0267677  0.6052060  0.45235747
33   -1.69142120  0.3853757 -0.9733521 -0.1263378 -1.06601373
34   -1.17846445  0.3853757 -0.9733521 -0.1263378 -1.06601373
35    0.01843465  0.3853757  1.0267677  1.3367497  0.45235747
36   -0.20954613 -2.5933307 -0.9733521 -0.8578816 -0.05376626
37    1.50030972  0.3853757  1.0267677  0.6052060 -1.57213746
38    0.64538179  0.3853757  1.0267677  0.6052060 -1.57213746
⋮    ⋮           ⋮          ⋮          ⋮          ⋮          
1969  0.36040582  0.3853757  1.0267677  2.0682935  0.45235747
1970 -1.17846445 -2.5933307 -0.9733521 -0.8578816  1.46460494
1971 -1.46344042  0.3853757 -0.9733521  1.3367497  0.95848121
1973  0.98735296  0.3853757  1.0267677  0.6052060 -0.55988999
1974 -1.46344042 -2.5933307  1.0267677 -0.8578816  1.46460494
1976 -1.12146925  0.3853757 -0.9733521  1.3367497 -0.55988999
1977  0.93035777  0.3853757  1.0267677 -0.1263378 -1.06601373
1978 -0.77949808 -2.5933307 -0.9733521 -0.8578816  0.45235747
1979 -1.29245484 -2.5933307 -0.9733521 -0.1263378 -1.06601373
1980  0.64538179  0.3853757  1.0267677 -0.8578816  0.45235747
1981  0.13242504  0.3853757 -0.9733521  2.7998373 -2.07826120
1982 -1.23545964  0.3853757 -0.9733521  1.3367497 -1.06601373
1983  0.64538179  0.3853757  1.0267677 -0.1263378 -0.05376626
1984 -0.32353652 -2.5933307  1.0267677  1.3367497 -1.06601373
1985 -1.69142120  0.3853757  1.0267677 -0.1263378 -0.55988999
1986 -0.72250289  0.3853757 -0.9733521 -0.1263378 -1.06601373
1987  0.75937218  0.3853757 -0.9733521 -0.8578816  1.46460494
1988  0.58838660  0.3853757  1.0267677 -0.8578816  1.46460494
1989 -0.83649328  0.3853757  1.0267677  1.3367497 -0.05376626
1990  0.41740101  0.3853757  1.0267677 -0.8578816 -0.55988999
1991  0.81636738  0.3853757  1.0267677 -0.8578816  0.45235747
1993 -0.09555574  0.3853757  1.0267677 -0.8578816 -0.55988999
1994  1.32932413  0.3853757  1.0267677  0.6052060 -0.05376626
1995  1.21533374  0.3853757  1.0267677  0.6052060  1.46460494
1996 -0.77949808 -2.5933307 -0.9733521 -0.8578816  0.45235747
1997 -0.55151730  0.3853757  1.0267677  1.3367497  0.45235747
1998  1.61430011  0.3853757 -0.9733521  0.6052060 -0.55988999
1999 -0.49452211  0.3853757 -0.9733521 -0.1263378  0.45235747
2000  0.30341062  0.3853757  1.0267677 -0.8578816  0.45235747
2001 -1.00747886 -2.5933307  1.0267677  1.3367497 -1.06601373

### Clustering data for k = 5

In [4]:
set.seed(76964057) #Set the seed for reproducibility
k1 <-kmeans(scaled_data, centers=5) 
k1$centers
table(k1$cluster)

Age         Date       Game        Empl       Education 
1  0.88611352  0.3853757  1.02676775  0.6919659 -0.1397118
2 -0.03520789  0.3853757  0.35153548 -0.7755595  0.8484543
3  0.66562719  0.3853757 -0.97335208  1.0288800 -0.1221614
4 -0.36013894 -2.5933307 -0.06504078 -0.1699620  0.1830439
5 -0.91674660  0.3853757 -0.56525589 -0.4100981 -0.6461353


  1   2   3   4   5 
371 391 259 218 446 

<b>Centers for clusters :<br>
Cluster 1 : 371<br>
Cluster 2 : 391<br>
Cluster 3 : 259<br>
Cluster 4 : 218<br>
Cluster 5 : 446<br>
<br>
Dominating features for clusters:<br>
Cluster 1 : Game (Playing video games on cell phone)<br>
Value : 1.02676775<br><br>
Cluster 2 : Education (Highest level of education completed or in progress)<br>
Value : 0.8484543<br><br>
Cluster 3 : Employment (Employed full time, part time, retired or not employed)<br>
Value : 1.0288800<br><br>
Cluster 4 : Education (Highest level of education completed or in progress)<br>
Value : 0.1830439<br><br>
Cluster 5 : Date (Usage of Online dating apps)<br>
Value : 0.3853757</b>

### Clustering data for k = 3

In [5]:
set.seed(76964057) #Set the seed for reproducibility
k2 <-kmeans(scaled_data, centers=3) 
k2$centers
table(k2$cluster)

Age        Date        Game       Empl       Education  
1  0.8752169  0.29356622  0.2219250  0.9095983 -0.06243276
2 -0.0546496 -0.38403189  0.3588311 -0.7067139  0.75958018
3 -0.8613713  0.06565762 -0.5797686 -0.2650563 -0.67125532


  1   2   3 
584 542 559 

<b>Centers for clusters :<br>
Cluster 1 : 548<br>
Cluster 2 : 569<br>
Cluster 3 : 568<br>
<br>
Dominating features for clusters:<br>
Cluster 1 : Education (Highest level of education completed or in progress)<br>
Value : 0.75958018<br><br>
Cluster 2 : Employment (Employed full time, part time, retired or not employed)<br>
Value : 0.9095983<br><br>
Cluster 3 : Date (Usage of Online dating apps)<br>
Value : 0.06565762</b>